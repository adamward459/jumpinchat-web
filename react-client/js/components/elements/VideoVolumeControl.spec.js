import React from 'react';
import { shallow } from 'enzyme';
import VideoVolumeControl from './VideoVolumeControl.react';

describe('<VideoVolumeControl />', () => {
  let props;
  beforeEach(() => {
    props = {
      volume: 0,
      showControl: false,
      onChange: jest.fn(),
      onSetControl: jest.fn(),
      onClickOutside: jest.fn(),
    };
  });

  describe('onChangeVolume', () => {
    it('should call onChange', () => {
      const wrapper = shallow(<VideoVolumeControl {...props} />);
      wrapper.instance().onChangeVolume(100);
      expect(props.onChange).toHaveBeenCalledWith(100);
    });
  });

  describe('render', () => {
    it('should not show volume control if showControl is false', () => {
      const wrapper = shallow(<VideoVolumeControl {...props} />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('should show volume control if showControl is true', () => {
      props.showControl = true;
      const wrapper = shallow(<VideoVolumeControl {...props} />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('should show correct icon when volume > 0', () => {
      props.volume = 100;
      const wrapper = shallow(<VideoVolumeControl {...props} />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });
});
