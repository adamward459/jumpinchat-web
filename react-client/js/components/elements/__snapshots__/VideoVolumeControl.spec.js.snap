// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoVolumeControl /> render should not show volume control if showControl is false 1`] = `
<TetherComponent
  attachment="top center"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "scrollParent",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
>
  <button
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "volume-off",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </button>
</TetherComponent>
`;

exports[`<VideoVolumeControl /> render should show correct icon when volume > 0 1`] = `
<TetherComponent
  attachment="top center"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "scrollParent",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
>
  <button
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "volume-up",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </button>
</TetherComponent>
`;

exports[`<VideoVolumeControl /> render should show volume control if showControl is true 1`] = `
<TetherComponent
  attachment="top center"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "scrollParent",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
>
  <button
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "volume-off",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </button>
  <div
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <ForwardRef
      onChangeVolume={[MockFunction]}
      volume={0}
    />
  </div>
</TetherComponent>
`;
